### PR TITLE
Created algorithm and factory for calculating hadronic final state variables

### PR DESCRIPTION
**Pull request by tylerkutz**
### Briefly, what does this PR introduce?

This adds and algorithm and corresponding factory for calculating hadronic final state variables.  Since this uses a new `EDM4eic` data type not included in major release 5, the algorithm is only called for major release >= 6.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue #1362 )
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### CODE REVIEW

File: src/algorithms/reco/HadronicFinalState.cc [hunk 1]
```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024 Tyler Kutz
+
```
Comment (veprbl) [inline src/algorithms/reco/HadronicFinalState.cc:3]:
```suggestion

#include <edm4eic/EDM4eicVersion.h>
#if EDM4EIC_VERSION_MAJOR >= 6
```

File: src/algorithms/reco/HadronicFinalState.cc [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    if (sigma <= 0) {
+      m_log->debug("Sigma zero or negative");
+      return;
+    }
+
+    m_log->debug("sigma_h, pT_h, gamma_h = {},{},{}", hfs.getSigma(), hfs.getPT(), hfs.getGamma());
+
+  }
+
+} // namespace Jug::Reco
```
Comment (veprbl) [inline src/algorithms/reco/HadronicFinalState.cc:150]:
```suggestion
} // namespace Jug::Reco
#endif
```

File: src/global/reco/reco.cc [hunk 3]
```diff
@@ -19,6 +20,7 @@
 #include "algorithms/reco/InclusiveKinematicsJB.h"
 #include "algorithms/reco/InclusiveKinematicsSigma.h"
 #include "algorithms/reco/InclusiveKinematicseSigma.h"
+#include "algorithms/reco/HadronicFinalState.h"
```
Comment (veprbl) [inline src/global/reco/reco.cc:23]:
```suggestion
#if EDM4EIC_VERSION_MAJOR >= 6
#include "algorithms/reco/HadronicFinalState.h"
#endif
```

File: src/global/reco/reco.cc [hunk 4]
```diff
@@ -27,6 +29,7 @@
 #include "factories/reco/InclusiveKinematicsTruth_factory.h"
 #include "factories/reco/JetReconstruction_factory.h"
 #include "factories/reco/TransformBreitFrame_factory.h"
+#include "factories/reco/HadronicFinalState_factory.h"
```
Comment (veprbl) [inline src/global/reco/reco.cc:32]:
```suggestion
#if EDM4EIC_VERSION_MAJOR >= 6
#include "factories/reco/HadronicFinalState_factory.h"
#endif
```

**Review by veprbl — COMMENTED** [review]
This could use more guards

**Comment by tylerkutz** [comment]
Thanks @veprbl , I agree that I had left out some necessary guards.  Have added them now (not yet pushed though).

I'm having a different problem now.  When I compile against my version of `EDM4eic` and lower the `EDM4EIC_VERSION_MAJOR` requirement to 5, the algorithm still isn't called...currently trying to figure out why.

**Comment by tylerkutz** [comment]
@veprbl Pushed with additional guards.  Passes build checks now.

**Comment by veprbl** [comment]
EDM4eic 6 should be in the nightly now, let's see how that fares.

**Review by veprbl — APPROVED** [review]

**Pull request merged**